### DISCUSSION TITLE
How do you implement Continuous batching of incoming requests?

### DISCUSSION THREAD

**Discussion by wqh17101** [vllm-project/vllm#492] [Q&A]
Where can i find the codes about it? I am really interested about it.

**Comment by irasin** [comment]
mark

**Comment by lucasjinreal** [comment]
Looks like doesn't support batch, for example

```
def forward(
        self,
        hidden_states: torch.Tensor,
        positions: torch.LongTensor,
        kv_cache: KVCache,
        input_metadata: InputMetadata,
        cache_event: Optional[torch.cuda.Event],
    ) -> torch.Tensor:
```

the Attention hiddenstate here has no batch dim

**Comment by zhuohan123** [comment]
Please check the logic starting at this `step` function:

https://github.com/vllm-project/vllm/blob/b4b195b3608610533e9d2c6c0168be71a8436355/vllm/engine/llm_engine.py#L228

Move this issue to discussions for future questions
